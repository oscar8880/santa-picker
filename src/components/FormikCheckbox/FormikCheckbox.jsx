// import styles from './FormikCheckbox.module.scss';
import React from 'react';
import PropTypes from 'prop-types';
// import classNames from 'classnames';
import { useField } from 'formik';

const FormikCheckbox = ({ children, ...props }) => {
  const [field, meta] = useField({ ...props, type: 'checkbox' });
  return (
    <>
      <label className="checkbox">
        <input type="checkbox" {...field} {...props} />
        {children}
      </label>
      {meta.touched && meta.error ? (
        <div className="error">{meta.error}</div>
      ) : null}
    </>
  );
};

FormikCheckbox.propTypes = {
  label: PropTypes.string,
  name: PropTypes.string,
  id: PropTypes.string.isRequired,
  children: PropTypes.oneOfType([PropTypes.string, PropTypes.element]),
};

export default FormikCheckbox;
