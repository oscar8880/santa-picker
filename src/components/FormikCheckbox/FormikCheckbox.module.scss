@import "../../scss/functions";
@import "../../scss/variables";

.FormikCheckbox {
  align-items: center;
  -webkit-appearance: none;
  border: 0;
  cursor: pointer;
  position: relative;
  text-decoration: none;
  transition: $transition-default;
}

.FormikCheckbox__Appearance--button-primary,
.FormikCheckbox__Appearance--button-secondary,
.FormikCheckbox__Appearance--button-primary-outline,
.FormikCheckbox__Appearance--button-secondary-outline {
  border-style: solid;
  border-width: 1px;
  display: inline-flex;
  font-family: $font-family-base;
  font-weight: bold;
  justify-content: center;
  line-height: rem(22px);
  padding: 10px 40px;
  text-align: center;
}

.FormikCheckbox__Appearance--button-primary {
  background-color: $primary-colour;
  border-color: $primary-colour;
  color: $white;

  &:hover:not(:disabled),
  &:focus:not(:disabled),
  &:active:not(:disabled) {
    background-color: darken($primary-colour, 5%);
    border-color: darken($primary-colour, 5%);
  }

  &:disabled {
    cursor: default;
    opacity: 0.5;
  }
}

.FormikCheckbox__Appearance--button-secondary {
  background-color: $secondary-colour;
  border-color: $secondary-colour;
  color: $white;

  &:disabled {
    cursor: default;
    opacity: 0.5;
  }

  &:hover:not(:disabled),
  &:focus:not(:disabled),
  &:active:not(:disabled) {
    background-color: darken($secondary-colour, 5%);
    border-color: darken($secondary-colour, 5%);
  }
}

.FormikCheckbox__Appearance--button-primary-outline,
.FormikCheckbox__Appearance--button-secondary-outline {
  background-color: transparent;
}

.FormikCheckbox__Appearance--button-primary-outline {
  border-color: $primary-colour;
  color: $primary-colour;

  &:hover,
  &:focus,
  &:active {
    background-color: $primary-colour;
    color: $white;
  }
}

.FormikCheckbox__Appearance--button-secondary-outline {
  border-color: $secondary-colour;
  color: $secondary-colour;

  &:hover,
  &:focus,
  &:active {
    background-color: $secondary-colour;
    color: $white;
  }
}

.FormikCheckbox__Appearance--a {
  background-color: transparent;
  color: #007aff;

  &:hover,
  &:focus,
  &:active {
    color: darken(#007aff, 5%);
  }
}

.FormikCheckbox__Appearance--none {
  appearance: none;
  background-color: transparent;
  border: 0;
  color: inherit;
  cursor: pointer;
  display: inline-block;
  font-size: inherit;
  line-height: inherit;
  padding: 0;
  text-align: inherit;
}

.FormikCheckbox__FullWidth {
  width: 100%;
}
