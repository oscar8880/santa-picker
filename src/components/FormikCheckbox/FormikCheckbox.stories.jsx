import React from 'react';
import FormikCheckbox from './FormikCheckbox';
import { Formik, Form } from 'formik';

export default {
  component: FormikCheckbox,
  title: 'FormikCheckbox',
};

export const Default = () => {
  return (
    <Formik initialValues={{}} onSubmit={() => {}}>
      <Form>
        <FormikCheckbox name="oi" label="Input" />
      </Form>
    </Formik>
  );
};
