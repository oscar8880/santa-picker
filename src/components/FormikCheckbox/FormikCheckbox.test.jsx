import React from 'react';
import { render } from '@testing-library/react';
import { Formik, Form } from 'formik';

import FormikCheckbox from './FormikCheckbox';

test('matches snapshot', async () => {
  const { asFragment } = render(
    <Formik initialValues={{}} onSubmit={() => {}}>
      <Form>
        <FormikCheckbox id="test" name="test" label="test" />
      </Form>
    </Formik>,
  );

  expect(asFragment()).toMatchSnapshot();
});
