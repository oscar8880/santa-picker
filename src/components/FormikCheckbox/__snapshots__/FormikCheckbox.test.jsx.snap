// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <form
    action="#"
  >
    <label
      class="checkbox"
    >
      <input
        id="test"
        label="test"
        name="test"
        type="checkbox"
        value=""
      />
    </label>
  </form>
</DocumentFragment>
`;
